lib/bunch/markdown.ex

```diff
@@ -0,0 +1,74 @@
+defmodule Bunch.Markdown do
+  @moduledoc """
+  A bunch of helpers for generating Markdown text
+  """
+
+  @doc """
+  Indents whole block of text by one level (two spaces).
+
+  ## Examples
+
+      iex>#{inspect(__MODULE__)}.indent("text")
+      "  text"
+
+      iex>text = \"""
+      ...>First line
+      ...>Second line
+      ...>Third line
+      ...>\"""
+      iex>#{inspect(__MODULE__)}.indent(text)
+      \"""
+        First line
+        Second line
+        Third line
+      \"""
+      iex>#{inspect(__MODULE__)}.indent(text, 2)
+      \"""
+          First line
+          Second line
+          Third line
+      \"""
+  """
+
+  @spec indent(String.t(), non_neg_integer()) :: String.t()
+  def indent(string, level \\ 1) do
+    # replace each line with indented one, `\0` is a whole match
+    string |> String.replace(~r/^.*$/m, indent_line("\\0", level))
```

Comment:
wouldn't it be simpler (and faster) to do `string |> String.split("\n") |> Enum.map(&indent_line(&1, level)) |> Enum.join("\n")`